README.md

```diff
@@ -92,14 +95,35 @@ class Genome(ABC):
 
 ```
 
-The `ABC` and `@abstractmethod` just means that this class is not something you can use by itself, but that another class must implement the details. In `src/genome.py` you will find templates for a Python list tand a linked list implementation (without the actual implementation, because you have to implement them).
+The `ABC` and `@abstractmethod` just means that this class is not something you can use by itself, but that another class must implement the details. In `src/genome.py` you will find templates for a Python list and a linked list implementation (without the actual implementation, because you have to implement them).
 
-You are free to implement the genome classes however you want, and using whateer auxilary data structures you desire, as long as one uses a Python list with an element for each nucleotide and the other a linked list with a link for each nucleotide. If you want to implement a third (or fourth or fifth...) version, you are very welcome to do so as well.
+You are free to implement the genome classes however you want, and using whatever auxilary data structures you desire, as long as one uses a Python list with an element for each nucleotide and the other a linked list with a link for each nucleotide. If you want to implement a third (or fourth or fifth...) version, you are very welcome to do so as well.
 
 ## Complexity
 
 When you have implemented the two (or more) classes, describe the complexity of each operation as a function of the genome size (at the time of the operation), and the size of the TE involved (and when copying, the offset you are copying). Put the description here:
 
 **FIXME: OPERATION COMPLEXITY**
 
+ListGenome:
+
+    - init(): The complexity is O(1)
+    - insert.te(): The complexity of this function depends on the length of the TE dict. O(TEn)
```

Comment:
If the new TE has length $m$, then `TE = ["A"] * TE_length` takes time $O(m)$. If there are $k$ active TEs, then the loop through them takes $O(k)$ (but $k\leq n$, so you can also say $O(n)$ if you want). Something like this `self.genome_list[:pos] + TE + self.genome_list[pos:]` takes time $O(n+m)$. The total time is $O(n+m)$.